holoviews/core/data/ibis.py

```diff
@@ -163,16 +168,16 @@ def histogram(cls, expr, bins, density=True, weights=None):
         else:
             # sort_by will be removed in Ibis 5.0
             hist_bins = binned.value_counts().sort_by('bucket').execute()
-
-        for b, v in zip(hist_bins['bucket'], hist_bins['count']):
+        metric_name = 'bucket_count' if ibis5() else 'count'
+        for b, v in zip(hist_bins['bucket'], hist_bins[metric_name]):
             if np.isnan(b):
                 continue
             hist[int(b)] = v
         if weights is not None:
             raise NotImplementedError("Weighted histograms currently "
                                       "not implemented for IbisInterface.")
         if density:
-            hist = hist/expr.count().execute()
+            hist = hist/expr.count().execute()/np.diff(bins)
```

Comment:
Can the bins be different sizes?

If they can, maybe add a small test where that is the case. 

If they can't, we can reduce the calculation by just dividing by the first value (`bins[1] - bins[0]`)

Reply:
Yes they can:
![image](https://github.com/holoviz/holoviews/assets/35924738/2a455f27-3e95-4fcf-9315-62772f92a214)

Added a test in https://github.com/holoviz/holoviews/pull/5929/commits/9e02079464c8d304d72894e1c71991df204aaf03